pgjdbc/src/test/java/org/postgresql/test/jdbc42/LargeCountJdbc42Test.java

```diff
@@ -0,0 +1,379 @@
+    try (Statement stmt = con.createStatement()) {
+      long count = stmt.executeLargeUpdate("delete from largetable");
+      Assert.assertEquals("Statement 110 rows more than Integer.MAX_VALUE", expected, count);
+    }
+    con.setAutoCommit(true);
+  }
+
+  /*
+   * Test Statement.executeLargeUpdate(String sql)
+   */
+  @Test
+  public void testExecuteLargeUpdateStatementSMALL() throws Exception {
+    try (Statement stmt = con.createStatement()) {
+      long count = stmt.executeLargeUpdate("insert into largetable "
+          + "select true from generate_series(1, 1010)");
+      long expected = 1010L;
+      Assert.assertEquals("Small long return 1010L", expected, count);
+    }
+  }
+
+  /*
+   * Test PreparedStatement.executeLargeUpdate();
+   */
+  @Test
+  public void testExecuteLargeUpdatePreparedStatementSMALL() throws Exception {
+    try (PreparedStatement stmt = con.prepareStatement("insert into largetable "
+        + "select true from generate_series(?, ?)")) {
+      stmt.setLong(1, 1);
+      stmt.setLong(2, 1010L);
+      long count = stmt.executeLargeUpdate();
+      long expected = 1010L;
+      Assert.assertEquals("Small long return 1010L", expected, count);
+    }
+  }
+
+  /*
+   * Test Statement.getLargeUpdateCount();
+   */
+  @Test
+  public void testGetLargeUpdateCountStatementSMALL() throws Exception {
+    try (Statement stmt = con.createStatement()) {
+      boolean isResult = stmt.execute("insert into largetable "
+          + "select true from generate_series(1, 1010)");
+      Assert.assertFalse("False if it is an update count or there are no results", isResult);
+      long count = stmt.getLargeUpdateCount();
+      long expected = 1010L;
+      Assert.assertEquals("Small long return 1010L", expected, count);
+    }
+  }
+
+  /*
+   * Test PreparedStatement.getLargeUpdateCount();
+   */
+  @Test
+  public void testGetLargeUpdateCountPreparedStatementSMALL() throws Exception {
+    try (PreparedStatement stmt = con.prepareStatement("insert into largetable "
+        + "select true from generate_series(?, ?)")) {
+      stmt.setInt(1, 1);
+      stmt.setInt(2, 1010);
+      boolean isResult = stmt.execute();
+      Assert.assertFalse("False if it is an update count or there are no results", isResult);
+      long count = stmt.getLargeUpdateCount();
+      long expected = 1010L;
+      Assert.assertEquals("Small long return 1010L", expected, count);
+    }
+  }
+
+  /*
+   * Test fail SELECT Statement.executeLargeUpdate(String sql)
+   */
+  @Test
+  public void testExecuteLargeUpdateStatementSELECT() throws Exception {
+    try (Statement stmt = con.createStatement()) {
+      long count = stmt.executeLargeUpdate("select true from generate_series(1, 5)");
+      Assert.fail("A result was returned when none was expected. Returned: " + count);
+    } catch (SQLException e) {
+      // TOO_MANY_RESULTS("0100E"), is this the correct SQLState?
+      Assert.assertEquals("0100E", e.getSQLState());
```

Comment:
It's better use `org.postgresql.util.PSQLState` enum values, however I `TOO_MANY_RESULTS` sounds strange for that case indeed

Reply:
Changed to use the PSQLState.

My question was because I have not found anywere this status code, it's not part of [PostgreSQL error codes](https://www.postgresql.org/docs/current/static/errcodes-appendix.html), and belongs to the _Class 01- Warning_ group, so I guess is made up just for the driver use but is just strange, this is how the driver handle it since long time so I don't touch that part.